app/models/user.rb

```diff
@@ -8,5 +8,21 @@ class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable
+         :recoverable, :rememberable, :trackable, :validatable,
+         :omniauthable, omniauth_providers: [:github, :facebook]
+
+  def self.from_omniauth(auth)
+    if User.where(email: auth.info.email)[0]
```

Comment:
`.find_by` is the equivalent of `.where( ...)[0]`. It will return the first instance that matches